ddtrace/tracer/tracer.go

```diff
@@ -306,16 +316,50 @@ func (t *tracer) worker(tick <-chan time.Time) {
 	}
 }
 
-func (t *tracer) pushTrace(trace []*span) {
+// traceInfo contains span list and sampling decision of the trace
+// needed to run single span sampling.
+type traceInfo struct {
+	spans            []*span
+	samplingDecision samplingDecision
+}
+
+// processTraceInfo runs single span sampling on the spans pushed to the worker thread.
+// Sampling spans on the worker thread keeps any associated latency off the application thread.
+func (t *tracer) processTraceInfo(info *traceInfo) {
```

Comment:
Speaking of my previous comment, here we find both the word "process" and "info" as part of the same method. Double whammy 🙂 How about `sampleFinishedTrace`? This makes it clear that it is sampling which occurs at the end.